e2e/snapshot_ops_test.go

```diff
@@ -0,0 +1,234 @@
+	"github.com/gluster/glusterd2/pkg/restclient"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	snapname = "snaptest"
+)
+
+// TestSnapshot creates a volume and snapshots, runs further tests on it and
+// finally deletes the volume
+func TestSnapshot(t *testing.T) {
+	var err error
+	var brickPaths []string
+	r := require.New(t)
+
+	gds, err = setupCluster("./config/1.toml", "./config/2.toml")
+	if err != nil {
+		r.Nil(err)
+		teardownCluster(gds)
+		return
+	}
+	defer teardownCluster(gds)
+
+	client = initRestclient(gds[0].ClientAddress)
+	brickPaths, err = lvmtest.CreateLvmBricks(4)
+	if err != nil {
+		r.Nil(err)
+		lvmtest.CleanupLvmBricks(4)
+		return
+	}
+	defer func() {
+		err := lvmtest.CleanupLvmBricks(4)
+		r.Nil(err)
+	}()
+
+	// Create the volume
+	if err := volumeCreateOnLvm(brickPaths, client); err != nil {
+		t.Logf("Failed to create the volume")
+		r.Nil(err)
+		return
+	}
+
+	defer func() {
+		err := volumeDelete(client)
+		r.Nil(err)
+	}()
+
+	if err := volumeStart(client); err != nil {
+		t.Logf("Failed to start the volume")
+		r.Nil(err)
+		return
+	}
+
+	defer func() {
+		err := volumeStop(client)
+		r.Nil(err)
+	}()
+
+	t.Run("Create", testSnapshotCreate)
+	t.Run("Activate", testSnapshotActivate)
+	t.Run("Deactivate", testSnapshotDeactivate)
+	t.Run("List", testSnapshotList)
+	t.Run("Info", testSnapshotInfo)
+	t.Run("Delete", testSnapshotDelete)
+
+	/*
+		TODO:
+		test snapshot on a volume that is expanded or shrinked
+	*/
+
+}
+
+func createLvmBricks(number int) ([]string, error) {
+	/*
+		if err := verifyLvm(); err != nil {
+			return err
+		}
+	*/
+	var brickPaths []string
+	for i := 1; i <= number; i++ {
+		brickPath, err := ioutil.TempDir(tmpDir, "brick")
+		if err != nil {
+			return brickPaths, err
+		}
+		brickPaths = append(brickPaths, brickPath)
+	}
+	return brickPaths, nil
+}
+
+func volumeCreateOnLvm(brickPaths []string, client *restclient.Client) error {
+
+	// create 2x2 dist-rep volume
+	createReq := api.VolCreateReq{
+		Name: volname,
+		Subvols: []api.SubvolReq{
+			{
+				ReplicaCount: 2,
+				Type:         "replicate",
+				Bricks: []api.BrickReq{
+					{PeerID: gds[0].PeerID(), Path: brickPaths[0]},
+					{PeerID: gds[1].PeerID(), Path: brickPaths[1]},
+				},
+			},
+			{
+				Type:         "replicate",
+				ReplicaCount: 2,
+				Bricks: []api.BrickReq{
+					{PeerID: gds[0].PeerID(), Path: brickPaths[2]},
+					{PeerID: gds[1].PeerID(), Path: brickPaths[3]},
+				},
+			},
+		},
+		Force: true,
+	}
+	_, err := client.VolumeCreate(createReq)
+	return err
+}
+
+func volumeDelete(client *restclient.Client) error {
+	return client.VolumeDelete(volname)
+}
+
+func testSnapshotCreate(t *testing.T) {
+	r := require.New(t)
+	snapshotCreateReq := api.SnapCreateReq{
+		VolName:  volname,
+		SnapName: snapname,
+	}
+	_, err := client.SnapshotCreate(snapshotCreateReq)
+	r.Nil(err, "snapshot create failed")
+
+	snapshotCreateReq = api.SnapCreateReq{
+		VolName:     volname,
+		SnapName:    snapname,
```

Comment:
use test case name t.Name() for volName and Snapname

Reply:
we could only use for one, both cannot exist with the same name